packages/mui-material/src/Autocomplete/Autocomplete.test.js

```diff
@@ -249,6 +249,48 @@ describe('<Autocomplete />', () => {
       expect(getByRole('listbox')).to.have.property('scrollTop', 0);
     });
 
+    // https://github.com/mui/material-ui/issues/36212
+    it('should preserve scrollTop position of the listbox when adding new options on mobile', function test() {
+      if (/jsdom/.test(window.navigator.userAgent)) {
+        this.skip();
+      }
+      function getOptions(count) {
+        return Array(count)
+          .fill('item')
+          .map((value, i) => value + i);
+      }
+
+      const { getByRole, getAllByRole, setProps } = render(
+        <Autocomplete
+          open
+          options={getOptions(5)}
+          renderInput={(params) => <TextField {...params} />}
+          ListboxProps={{ style: { padding: 0, maxHeight: '100px' } }}
+          PopperComponent={(props) => {
+            const { disablePortal, anchorEl, open, ...other } = props;
+            return <Box {...other} />;
+          }}
+        />,
+      );
+      const textbox = getByRole('combobox');
+      const listbox = getByRole('listbox');
+
+      act(() => {
+        textbox.focus();
+      });
```

Comment:
```suggestion
```

I believe this can be tested without focusing the textbox because the listbox is already opened.

Reply:
Thanks